src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertProvider.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+
+        private sealed class ClientCertProvider : IDisposable
+        {
+            internal readonly string _certificateFile;
+            internal readonly string _password;
+
+            private ClientCertProvider(X509Certificate2Collection certs)
+            {
+                Debug.Assert(certs.Count >0);
+                try
+                {
+                    _password = GeneratePassword();
+                    byte[] pfxData = certs.Export(X509ContentType.Pkcs12, _password);
+                    _certificateFile = Path.GetTempFileName();
```

Comment:
For the callback approach, if the only way to figure out is setopt failure, can we cache that in the static ctor? In the default Ubuntu distro, this is going to fail always.